org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -20,10 +20,52 @@ public class NearInteractionModeDetector : ProximityDetector
         [Tooltip("The set of near interactors that belongs to near interaction")]
         private List<XRBaseInteractor> nearInteractors;
 
+        /// <summary>
+        /// Used to keep track of the previously detected colliders so that we can know which
+        /// colliders stopped being detected and update their buttons front plate and rounded
+        /// rect if they are labeled as dynamic (based on proximity).
+        /// </summary>
+        List<Collider> previouslyDetectedColliders = new List<Collider>();
+
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
-            return base.IsModeDetected() || IsNearInteractorSelecting();
+            bool result = base.IsModeDetected() || IsNearInteractorSelecting();
+            if (result)
+            {
+                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                {
+                    Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
+                    if (!DetectedColliders.Contains(previouslyDetectedCollider) && previouslyDetectedCollider != null)
+                    {
+                        IXRProximityInteractable nearInteractionMode = previouslyDetectedCollider.GetComponent<IXRProximityInteractable>();
+                        if (nearInteractionMode != null)
+                        {
+                            foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
+                            {
+                                previouslyDetectedCollider.GetComponent<IXRProximityInteractable>().OnProximityExited(new ProximityExitedEventArgs(this, previouslyDetectedCollider, xrBaseInteractor));
+                            }
+                        }
+                        previouslyDetectedColliders.Remove(previouslyDetectedCollider);
+                    }
+                }
+                foreach (Collider collider in DetectedColliders)
+                {
+                    if (!previouslyDetectedColliders.Contains(collider))
+                    {
+                        IXRProximityInteractable nearInteractionMode = collider.GetComponent<IXRProximityInteractable>();
+                        if (nearInteractionMode != null)
+                        {
+                            foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
```

Comment:
Instead of sending multiple events to account for all `nearInteractors`, should we just send _one_ relaying info about either:

1. The `ProximityDetector` itself. We could potentially expose the list of `nearInteractors` as an `IReadOnlyList` or something if we think the interactables need that info
2. Specifically the `nearInteractor`(s) doing the selecting. They likely aren't all selecting at once, but one of the cases for this logic is that _any_ `nearInteractor` is selecting (`IsNearInteractorSelecting()`)